hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSink.java

```diff
@@ -107,6 +107,14 @@ public class ReplicationSink {
   private SourceFSConfigurationProvider provider;
   private WALEntrySinkFilter walEntrySinkFilter;
 
+  /**
+   * If enabled at sink cluster site config, extended WAL attributes would be attached as Mutation
+   * attributes. This is useful for source cluster coproc to provide coproc specific metadata as WAL
+   * annotations and have them attached back to Mutations generated from WAL entries at sink side.
+   */
+  public static final String HBASE_REPLICATION_SINK_ATTRIBUTES_WAL_TO_MUTATIONS =
```

Comment:
This doesn't need to be configurable. Mutation attributes will be ignored if they are not expected so this is harmless to clients. And IMHO the right thing to be doing. I agree with your premise.